#170873930 Fix the requests search data for the line manager

gildniy opened the pull request:
#### What does this PR do?
Fix the requests search data for the line manager
#### Description of Task to be completed?
- [x] Get all requests under manager report whether open or closed upon search
#### How should this be manually tested?
- clone this repo and open the project 
- install all dependencies using `npm install`
- in postman login as a line manager and go to `/requests/search?byLineManager=true?searchString=open`
#### Any background context you want to provide?
This bug fix task was needed to be done to allow the Line manager to search all request types (all, open and declined) and use the search terms (requester names, travel date, return date, city or country).
#### What are the relevant pivotal tracker stories?
[#170873930](https://www.pivotaltracker.com/story/show/170873930)
#### Screenshots (if appropriate)
N/A
#### Questions:
N/A

RaymondMwaura commented on src/utils/schemas.js:
```diff
@@ -112,18 +112,23 @@ const hotelSchema = Joi.object().keys({
   leavingFrom: Joi.number().strict(),
   goingTo: Joi.number().strict(),
   travelDate: Joi.date(),
-  returnDate: Joi.date().when('type', { is: 'return', then: Joi.required() }).allow(null, ''),
```
Ask @b0nbon1 about the omission of '.allow(null, '')' since I think he needs it for his update trips feature

RaymondMwaura commented on src/services/request.service.js:
```diff
@@ -76,12 +76,7 @@ const getOneRequest = async (requestId) => {
-      model: db.comment,
-      include: [{
-        model: db.user,
-        as: 'author',
```
I think @Mastel22 will need this for her comments feature. Ask her if you should omit it.

nshaphan committed:
bug(requests) Fix the requests search data
 - Updated the requests data returned to the Line Manager upon searching
[Finishes #170873930]

nshaphan reviewed (dismissed):
LGTM

RaymondMwaura reviewed (approved):
LGTM

b0nbon1 reviewed (approved):
LGTM
